tfworker/providers/aws.py

```diff
@@ -171,10 +218,28 @@ def delete_with_versions(config, key):
     note: in initial testing this isn't required, but is inconsistent with how S3 delete markers, and the boto
     delete object call work there may be some configurations that require extra handling.
     """
-    s3_client = config.session.client("s3")
+    s3_client = config.state_session.client("s3")
     s3_client.delete_object(Bucket=config.state_bucket, Key=key)
 
 
+def get_assumed_role_session(
+    session, role_arn, session_name="AssumedRoleSession1", duration=3600
+):
+    """ get_assumed_role_session returns a boto3 session updated with assumed role credentials """
+    sts_client = session.client("sts")
+    role_creds = sts_client.assume_role(
+        RoleArn=role_arn, RoleSessionName=session_name, DurationSeconds=duration
+    )["Credentials"]
+
+    new_session = boto3.Session(
```

Comment:
You might have probably taken care of the region elsewhere. 

But, it might be worth adding the `region_name` parameter to boto3.Session here similar to https://github.com/ephur/terraform-worker/pull/16/files#diff-5a81c63b31ad94e6e0578cbea55505a8R72 for consistency sake.